### PR TITLE
Enable --wasm flag for production builds

### PR DESCRIPTION
**Pull request by johnpryan**
The UI is working nicely when built with the latest version Flutter engine, so it would be great to start using Wasm for the production version of DartPad. (The remaining issues with platform views have been fixed on the main channel of the SDK.)

Here's a staged version built with the `--wasm` flag, using the latest Flutter SDK (main channel): https://dart-pad-a3c64--wasm-cdjiko6q.web.app/

I made sure to update the Flutter SDK images to the latest version of the `main` channel of the SDK in GCP, so once this is merged it should use those images.

### CODE REVIEW

**Comment by parlough** [comment]
Cool. Thanks!

> Here's a staged version built with the `--wasm` flag, using the latest Flutter SDK (main channel): https://dart-pad-a3c64.web.app/

I'm able to run other Flutter web apps with Wasm, but the staged version at the link you sent seems to still be running with (or picking) JS.

**Comment by johnpryan** [comment]
Sorry, wrong link: https://dart-pad-a3c64--wasm-cdjiko6q.web.app/

**Comment by johnpryan** [comment]
It looks like the platform view is still blank when changing between different samples:

dartpad 2024-08-08 13_57_42

**Comment by johnpryan** [comment]
This doesn't appear to be related to Wasm, but seems to be happening due to a timing issue related to calling refresh() on the Codemirror API. I patched this issue by adding an artificial delay whenever a new CodeMirror platform view is created: https://github.com/dart-lang/dart-pad/pull/3039/commits/75542ec212f25fa5f4cb02eabce0e92a19bae990

**Comment by johnpryan** [comment]
Updated staging link: https://dart-pad-a3c64--wasm-cdjiko6q.web.app/

File: pkgs/dartpad_ui/lib/editor/editor.dart [hunk 1]
```diff
@@ -199,7 +199,8 @@ class _EditorWidgetState extends State<EditorWidget> implements EditorService {
     // darkmode
     _updateCodemirrorMode(darkMode);
 
-    Timer.run(() => codeMirror!.refresh());
+    Future.delayed(
```
Comment (devoncarew) [inline pkgs/dartpad_ui/lib/editor/editor.dart:202]: We should add a brief comment for why we have this delay.

**Comment by devoncarew** [comment]
With this link: https://dart-pad-a3c64--wasm-cdjiko6q.web.app/, I don't see content in the codemirror editor until I click inside it.

**Comment by johnpryan** [comment]
>  I don't see content in the codemirror editor until I click inside it.
> We should add a brief comment for why we have this delay.

https://github.com/dart-lang/dart-pad/pull/3039/commits/63b1d9dbd95bd217f45155e3c27fb34c319b764f should fix that.

Updated staging link: https://sketch-pad-1cebb--wasm-x4ieryp1.web.app/

**Review by devoncarew — APPROVED** [review]
🎉

**Pull request merged**